packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -104,6 +137,72 @@ const modifiers = [
   },
 ];
 
+const ListButton = ({ icon, format, label }) => {
+  const editor = useSlate();
+
+  const isListActive = () => {
+    const { selection } = editor;
+
+    if (!selection) return false;
+
+    const [match] = Array.from(
+      Editor.nodes(editor, {
+        at: Editor.unhangRange(editor, selection),
+        match: (node) =>
+          !Editor.isEditor(node) &&
+          Element.isElement(node) &&
+          node.type === 'list' &&
```

Comment:
I think you do this exact check twice, is it something we will do often? Do you think it could be a helper function or something?